src/Process/3dTilesProcessing.js

```diff
@@ -7,41 +9,86 @@ function requestNewTile(view, scheduler, geometryLayer, metadata, parent) {
+        // additive refinement only fetch visible children
+        subdivideNodeAdditive(context, layer, node, cullingTest);
+    } else {
+        // substractive refinement on the other hand requires to replace
+        // node with all of its children
+        subdivideNodeSubstractive(context, layer, node);
+    }
+}
+
+function subdivideNodeAdditive(context, layer, node, cullingTest) {
+    const tmp = new THREE.Matrix4();
+    for (const child of layer.tileIndex.index[node.tileId].children) {
+        // child being downloaded => skip
+        if (child.promise || child.loaded) {
+            continue;
+        }
+
+        // 'child' is only metadata (it's *not* a THREE.Object3D). As cullingTest
+        // expect a matrixWorld, we compute it (thanks to parent's transformation and
+        // metadata.worldFromLocalTransform).
+        let overrideMatrixWorld;
+        if (child.worldFromLocalTransform) {
+            tmp.multiplyMatrices(node.matrixWorld, child.worldFromLocalTransform);
+            overrideMatrixWorld = tmp;
+        } else {
+            overrideMatrixWorld = node.matrixWorld;
+        }
+
+        const isVisible = cullingTest ? !cullingTest(child, context.camera, overrideMatrixWorld) : true;
+
+        // child is not visible => skip
+        if (!isVisible) {
+            continue;
+        }
+        child.promise = requestNewTile(context.view, context.scheduler, layer, child, node, true).then((tile) => {
+            node.add(tile);
+            tile.updateMatrixWorld();
+            context.view.notifyChange(true);
+            child.loaded = true;
+            delete child.promise;
+        });
+    }
+}
+
+function subdivideNodeSubstractive(context, layer, node) {
     if (!node.pendingSubdivision && node.children.filter(n => n.layer == layer.id).length == 0) {
         const childrenTiles = layer.tileIndex.index[node.tileId].children;
         if (childrenTiles === undefined || childrenTiles.length === 0) {
             return;
         }
-
         node.pendingSubdivision = true;
 
         const promises = [];
         for (let i = 0; i < childrenTiles.length; i++) {
             promises.push(
-                requestNewTile(context.view, context.scheduler, layer, childrenTiles[i], node).then((tile) => {
+                requestNewTile(context.view, context.scheduler, layer, childrenTiles[i], node, false).then((tile) => {
+                    childrenTiles[i].loaded = true;
                     node.add(tile);
                     tile.updateMatrixWorld();
                     if (node.additiveRefinement) {
                         context.view.notifyChange(true);
                     }
                 }));
         }
-
         Promise.all(promises).then(() => {
             node.pendingSubdivision = false;
             context.view.notifyChange(true);
         });
     }
 }
 
-export function $3dTilesCulling(node, camera) {
+export function $3dTilesCulling(node, camera, overrideMatrixWorld) {
```

Comment:
Not a big fan of this override. I would prefer something like this instead: 
```js
export function $3dTilesCulling(viewerRequestVolume, boundingVolume, matrixWorld, camera)
```

Reply:
I changed the function, and fixed the special case of boundingVolume.region. Let me know what you think.